modules/openapi-generator/src/main/resources/tiny-cpp-client/service/api-body.mustache

```diff
@@ -0,0 +1,236 @@
+        >
+        {{classname}}::
+        {{! Method name }}
+        {{nickname}}(
+        {{! Params }}
+        {{#allParams}}
+        {{! Arrays }}
+            {{#isContainer}}{{{dataType}}}<{{baseType}}> {{paramName}}{{/isContainer}}{{#isMap}}// TODO: Implement Maps{{/isMap}}
+            {{! Normal types/objects }}
+            {{^isContainer}}{{{dataType}}} {{paramName}}{{/isContainer}}
+            {{^-last}}, {{/-last}}
+        {{/allParams}}
+        )
+        {{! Method start}}
+        {
+            std::string url = basepath + "{{{path}}}"; //{{#pathParams}}{{{paramName}}} {{/pathParams}}
+            // Query    | {{#queryParams}}{{paramName}} {{/queryParams}}
+            // Headers  | {{#headerParams}}{{paramName}} {{/headerParams}}
+            // Form     | {{#formParams}}{{paramName}} {{/formParams}}
+            // Body     | {{#bodyParam}}{{paramName}}{{/bodyParam}}
+
+            {{#pathParams}}
+                string s_{{paramName}}("{");
+                s_{{paramName}}.append("{{{baseName}}}");
+                s_{{paramName}}.append("}");
+
+                int pos = url.find(s_{{paramName}});
+
+                url.erase(pos, s_{{paramName}}.length());
+                url.insert(pos, stringify({{paramName}}));
+            {{/pathParams}}
+
+            HTTPClient http;
+            {{#isESP32}}
+                http.begin(String(url.c_str()), test_root_ca); //HTTPS connection
+            {{/isESP32}}
+            {{^isESP32}}
+                http.begin(String(url.c_str())); //HTTP connection
+            {{/isESP32}}
+
+
+            std::string payload = "";
+            // Send Request
+            // METHOD | {{httpMethod}}
+            {{#bodyParam}}
+            http.addHeader("Content-Type", "application/json");
+
+            {{#isContainer}}
+            {{#isArray}}
+            {{#items}}
+            {{#isPrimitiveType}}
+            bourne::json tmp_arr = bourne::json::array();
+            for(auto& var : {{paramName}})
+            {
+                tmp_arr.append(var);
+            }
+            payload = tmp_arr.dump();
+            {{/isPrimitiveType}}
+
+            {{^isPrimitiveType}}
+            bourne::json tmp_arr = bourne::json::array();
+            for(auto& var : {{paramName}})
+            {
+                auto tmp = var.toJson();
+                tmp_arr.append(tmp);
+
+            }
+            payload = tmp_arr.dump();
+            {{/isPrimitiveType}}
+            {{/items}}
+            {{/isArray}}
+            {{/isContainer}}
+
+            {{^isContainer}}
+            {{#isPrimitiveType}}
+            payload = stringify({{paramName}});
+            {{/isPrimitiveType}}
+
+            {{^isPrimitiveType}}
+            payload = {{paramName}}.toJson().dump();
+            {{/isPrimitiveType}}
+            {{/isContainer}}
+
+            {{/bodyParam}}
+            int httpCode = http.sendRequest("{{httpMethod}}", (uint8_t *) payload.c_str(), payload.length());
```

Comment:
Another C-style cast. I guess it should become reinterpret_cast.

Reply:
We could't get reinterpret_cast<uint8_t*>() to work. We've copied what the underlying library does like this:
https://github.com/espressif/arduino-esp32/blob/a1d8b959b05e3fb1391883ce6f9f2ed87af7da0f/libraries/HTTPClient/src/HTTPClient.cpp#L513-L516

Reply:
Probably because you're casting away const from `const char* c_str()`. You can use `payload.data()`, take a look at std::string specification and see what's the difference between c_str and data.

https://en.cppreference.com/w/cpp/string/basic_string/c_str
https://en.cppreference.com/w/cpp/string/basic_string/data